spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java

```diff
@@ -62,31 +62,43 @@ public boolean isProvided(ProcedureParameter param) {
     return !args.isNullAt(ordinal);
   }
 
-  public boolean bool(ProcedureParameter param, boolean defaultValue) {
+  public Boolean asBoolean(ProcedureParameter param, Boolean defaultValue) {
     validateParamType(param, DataTypes.BooleanType);
     int ordinal = ordinal(param);
-    return args.isNullAt(ordinal) ? defaultValue : args.getBoolean(ordinal);
+    return args.isNullAt(ordinal) ? defaultValue : (Boolean) args.getBoolean(ordinal);
```

Comment:
An explicit cast is needed to prevent a NPE, probably same as [here](https://stackoverflow.com/questions/25417438/java-autoboxing-and-ternary-operator-madness).

Reply:
I cannot reproduce the NPE locally. Run this script on Java8 and Java11. Neither triggered the NPE. Maybe Java has fixed the issue?
```
jshell> int a = 1
a ==> 1

jshell> Boolean b = a== 1 ? null: true
b ==> null
```

Reply:
Looks like the literal `null` is special. It doesn't unbox. +1 for your solution. 
```
Boolean bv = null;
Boolean b = true? bv: true;  //throw NPE
Boolean b = true? bv: (Boolean)true;  // not throw NPE
```